nav2_controller/src/controller_server.cpp

```diff
@@ -451,17 +451,25 @@ void ControllerServer::computeControl()
 
     last_valid_cmd_time_ = now();
     rclcpp::WallRate loop_rate(controller_frequency_);
+    bool cancelling = false;
```

Comment:
I think having a cancelling timeout would be good so that if a controller is not implemented well, we can still stop it eventually. Default to like 5s but you can increase it for vehicle